From 80cc2da52e839916f9290b53a0540a29cb6442bc Mon Sep 17 00:00:00 2001
From: Petr Kovar <pknbe@volny.cz>
Date: Mon, 6 Dec 2021 23:21:21 +0100
Subject: [PATCH] Actually build with enchant-2 for GTK 3

CI: Don't bother trying to build with gtkspell-2.0 if enchant is
no longer ailable on Ubuntu.

CI: Since GTK 3 is not default, don't run make distcheck for it.
---
 .gitlab-ci.yml                | 11 +++++------
 configure.ac                  |  3 ++-
 pan/gui/group-prefs-dialog.cc |  4 ++++
 3 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index a719895..27e39ca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -59,6 +59,7 @@ GTK3_REQUIRED=3.0.0
 GTKSPELL_REQUIRED=2.0.7
 GTKSPELL3_REQUIRED=2.0.16
 ENCHANT_REQUIRED=1.6.0
+ENCHANT2_REQUIRED=2.2.3
 GNUTLS_REQUIRED=3.0.0
 LIBNOTIFY_REQUIRED=0.4.1
 LIBGKR_REQUIRED=3.2.0
@@ -159,7 +160,7 @@ if test "x$want_gtk3" = "xyes" ; then
                     AC_DEFINE(HAVE_GTK,[1],[GTK+ 3 support])]
                     )
   if test "x$want_gtkspell" = "xyes" ; then
-    PKG_CHECK_MODULES([GTKSPELL], [gtkspell3-3.0 >= $GTKSPELL3_REQUIRED enchant >= $ENCHANT_REQUIRED],
+    PKG_CHECK_MODULES([GTKSPELL], [gtkspell3-3.0 >= $GTKSPELL3_REQUIRED enchant-2 >= $ENCHANT2_REQUIRED],
                       [
                       gtkspell_msg=yes
                       AC_DEFINE(HAVE_GTKSPELL,[1],[GtkSpell support for spellchecking])
diff --git a/pan/gui/group-prefs-dialog.cc b/pan/gui/group-prefs-dialog.cc
index 8bd07a8..5b9296a 100644
--- a/pan/gui/group-prefs-dialog.cc
+++ b/pan/gui/group-prefs-dialog.cc
@@ -22,8 +22,12 @@
 #include <glib.h>
 #include "gtk-compat.h"
 #ifdef HAVE_GTKSPELL
+#if GTKSPELL_VERSION == 3
+#include <enchant-2/enchant.h>
+#else
 #include <enchant/enchant.h>
 #endif
+#endif
 #include <pan/general/debug.h>
 #include <pan/general/macros.h>
 #include <pan/data/data.h>
-- 
GitLab

